.github/workflows/main.yml

```diff
@@ -254,58 +66,11 @@ jobs:
         mingw-w64-cross-gcc-stage1,
         mingw-w64-cross-crt,
       ]
-    runs-on: windows-latest
-
-    steps:
-      - uses: msys2/setup-msys2@v2
-        with:
-          msystem: MSYS
-          location: ${{ github.workspace }}
-          release: true
-          update: true
-          cache: false
-          install: base-devel
-
-      - name: Checkout MSYS2 packages repository
-        uses: actions/checkout@v4
-        with:
-          repository: ${{ env.MSYS2_REPO }}
-          ref: ${{ env.MSYS2_BRANCH }}
-          path: ${{ github.workspace }}/MSYS2-packages
-
-      - name: Download mingw-w64-cross-headers
-        uses: actions/download-artifact@v4
-        with:
-          name: mingw-w64-cross-headers
-
-      - name: Download mingw-w64-cross-binutils
-        uses: actions/download-artifact@v4
-        with:
-          name: mingw-w64-cross-binutils
-
-      - name: Download mingw-w64-cross-gcc-stage1
-        uses: actions/download-artifact@v4
-        with:
-          name: mingw-w64-cross-gcc-stage1
-
-      - name: Download mingw-w64-cross-crt
-        uses: actions/download-artifact@v4
-        with:
-          name: mingw-w64-cross-crt
-
-      - name: Install artifacts
-        run: pacman -U --noconfirm *.pkg.tar.zst
-
-      - name: Build mingw-w64-cross-winpthreads
-        working-directory: ${{ github.workspace }}/MSYS2-packages/mingw-w64-cross-winpthreads
-        run: makepkg --syncdeps --rmdeps --cleanbuild --noconfirm --noprogressbar --force
-
-      - name: Upload mingw-w64-cross-winpthreads
-        uses: actions/upload-artifact@v4
-        with:
-          name: mingw-w64-cross-winpthreads
-          retention-days: 1
-          path: ${{ github.workspace }}/MSYS2-packages/mingw-w64-cross-winpthreads/*.pkg.tar.zst
+    uses: ./.github/workflows/build-package.yml
+    with:
+      package_name: mingw-w64-cross-winpthreads
```

Comment:
I would support build for multiple packages with one workflow call if it is possible.

Reply:
Can you be more specific what it would be good for and how would you prefer to do it? Note that different pacakages may have different source repository (`MINGW-packages`, `MSYS2-packages`), set of dependencies (`needs`, `dependencies`) or additional parameters that controls the build that might be introduced to this workflow in the future. If build of multiple packages should be handled in this workflow (in opposite to some parent workflow), it would require quite complicated structure to specify and parse for the parameters.

Reply:
I am not sure how long it takes to prepare env for every package build. It was just an idea for you to think about optimization if it is possible. Current implementation is good for me.

Reply:
If that's the point, I can say that preparing the env takes few minutes (3-5) and could be even optimized and the main reason why it is prepared for each package individually is to have stable, isolated and reproducible builds for each package.